src/server/test-suites/test-suite/proxy/proxy.test.js

```diff
@@ -0,0 +1,207 @@
+jest.mock('~/src/server/common/helpers/fetch/fetch-tenant-service.js')
+jest.mock('~/src/server/services/helpers/fetch/fetch-repository.js')
+jest.mock('~/src/server/test-suites/helpers/fetch/fetch-test-suite.js')
+jest.mock('~/src/server/common/helpers/fetch/fetch-json.js')
+jest.mock('~/src/server/common/helpers/auth/get-user-session.js')
+jest.mock('~/src/server/services/helpers/fetch/fetch-proxy-rules.js')
+
+describe('Proxy Test Suite page', () => {
+  /** @type {import('@hapi/hapi').Server} */
+  let server
+
+  beforeAll(async () => {
+    jest
+      .mocked(fetchProxyRules)
+      .mockImplementation((serviceName, environment) => {
+        return {
+          environment,
+          serviceName,
+          defaultDomains: environment !== 'prod' ? ['https://google.com'] : [],
+          allowedDomains: environment !== 'prod' ? ['https://abc.com'] : []
+        }
+      })
+
+    mockCommonTestSuiteCalls(jest, 'mock-test-suite')
+    server = await initialiseServer()
+  })
+
+  afterAll(async () => {
+    await server.stop({ timeout: 0 })
+  })
+
+  describe('all envs view', () => {
+    test('page renders for logged in admin user', async () => {
+      mockAuthResponse(jest, true, true, true)
+
+      const { result, statusCode } = await server.inject({
+        method: 'GET',
+        url: '/test-suites/mock-test-suite/proxy',
+        auth: {
+          credentials: {
+            user: { id: '123', displayName: 'My Test User' },
+            scope: [scopes.admin, scopes.tenant]
+          },
+          strategy: 'default'
+        }
+      })
+      expect(statusCode).toBe(200)
+      expect(result).toMatchFile()
+    })
+
+    test('page renders for logged in tenant', async () => {
+      mockAuthResponse(jest, false, true, true)
+
+      const { result, statusCode } = await server.inject({
+        method: 'GET',
+        url: '/test-suites/mock-test-suite/proxy',
+        auth: {
+          credentials: {
+            user: { id: '123', displayName: 'My Test User' },
+            scope: [scopes.tenant]
+          },
+          strategy: 'default'
+        }
+      })
+      expect(statusCode).toBe(200)
+      expect(result).toMatchFile()
+    })
+
+    test('page renders for logged in service owner tenant', async () => {
+      mockAuthResponse(jest, false, true, true, 'mock-team-id')
+
+      const { result, statusCode } = await server.inject({
+        method: 'GET',
+        url: '/test-suites/mock-test-suite/proxy',
+        auth: {
+          credentials: {
+            user: { id: '123', displayName: 'My Test User' },
+            scope: [scopes.admin, scopes.tenant, 'mock-team-id']
+          },
+          strategy: 'default'
+        }
+      })
+      expect(statusCode).toBe(200)
+      expect(result).toMatchFile()
+    })
+
+    test('page errors with 401 for logged out user', async () => {
+      mockAuthResponse(jest, undefined, undefined, false)
```

Comment:
so this is properly ugly. can I suggest destructuring an options object. This way you only pass it wheat you need and don't have to fill args with empty values.


Simply wrap the args with an object in the function you are calling:

```
function mockAuthResponse({
  jest,
  isAdmin,
  isTenant,
  isAuthenticated,
  teamScope = 'some-other-team-id',
  additionalScopes = []
}) {
...
```

And from where you are calling it do:

```
mockAuthResponse({ jest, isAuthenticated: false })
```

It also means:
- Args are documented, great especially when there are loads
- Don't have to pass in empty values to reach an arg further down 
- The names of variables stays the same outside and inside functions